sycl/include/CL/sycl/device.hpp

```diff
@@ -192,6 +195,10 @@ class __SYCL_EXPORT device {
   shared_ptr_class<detail::device_impl> impl;
   device(shared_ptr_class<detail::device_impl> impl) : impl(impl) {}
 
+  static std::unordered_map<cl_device_id, std::weak_ptr<detail::device_impl>>
+      device_impls;
```

Comment:
Is it possible to reuse already existing cache, see platform_impl.hpp:199 (MDeviceCache) ?

Reply:
I pushed another solution with reusing an existing code